Failing macOS build - Use native libuuid on *BSD

Hello,

I often build Gerbera on **Fedora**, but occasionally on **macOS**.  It seems like my macOS fails to compile after the FreeBSD libuuid commit --> 5fa52b3fdd89c7d55173aef42b500282cee9cbb3.

> If I revert the 5fa52b3fdd89c7d55173aef42b500282cee9cbb3, all is well.  

It is almost like `#ifdef BSD` is not respected during build time, but printing out the flag yields `BSD` is disabled.

I am novice with C++ so perhaps the **correct** setting evades my knowledge.

Build Options
```
-DWITH_MAGIC=1 -DWITH_MYSQL=0 
-DWITH_CURL=1 -DWITH_JS=0 -DWITH_TAGLIB=1 -DWITH_AVCODEC=1 
-DWITH_EXIF=1 -DWITH_LASTFM=0 -DWITH_INOTIFY=0
```

Build Output
```
/usr/local/bin/cmake --build /Users/gerbera/cmake-build-debug --target gerbera -- -j 4
Scanning dependencies of target libgerbera
[  1%] Building CXX object CMakeFiles/libgerbera.dir/src/config_manager.cc.o
[  2%] Building CXX object CMakeFiles/libgerbera.dir/src/tools.cc.o
In file included from /Users/gerbera/src/config_manager.cc:39:
/usr/local/Cellar/ossp-uuid/1.6.2_2/include/ossp/uuid.h:94:24: error: typedef redefinition with different types ('struct uuid_st' vs '__darwin_uuid_t' (aka 'unsigned char [16]'))
typedef struct uuid_st uuid_t;
                       ^
/usr/include/sys/_types/_uuid_t.h:30:25: note: previous definition is here
typedef __darwin_uuid_t uuid_t;
                        ^
/Users/gerbera/src/tools.cc:358:5: error: use of undeclared identifier 'uuid_create'
    uuid_create(&uuid, &status);
    ^
/Users/gerbera/src/tools.cc:359:5: error: use of undeclared identifier 'uuid_to_string'
    uuid_to_string(&uuid, &uuid_str, &status);
    ^
/Users/gerbera/src/config_manager.cc:2114:9: error: no matching function for call to 'uuid_create'
        uuid_create(&uuid, &status);
        ^~~~~~~~~~~
/usr/local/Cellar/ossp-uuid/1.6.2_2/include/ossp/uuid.h:97:22: note: candidate function not viable: requires single argument '_uuid', but 2 arguments were provided
extern uuid_rc_t     uuid_create   (      uuid_t **_uuid);
                     ^
/Users/gerbera/src/config_manager.cc:2115:9: error: use of undeclared identifier 'uuid_to_string'
        uuid_to_string(&uuid, &uuid_str, &status);
        ^
2 errors generated.
make[3]: *** [CMakeFiles/libgerbera.dir/src/tools.cc.o] Error 1
make[3]: *** Waiting for unfinished jobs....
3 errors generated.
make[3]: *** [CMakeFiles/libgerbera.dir/src/config_manager.cc.o] Error 1
make[2]: *** [CMakeFiles/libgerbera.dir/all] Error 2
make[1]: *** [CMakeFiles/gerbera.dir/rule] Error 2
make: *** [gerbera] Error 2
```

Further down the macOS rabbit hole.  Getting a linkage error...It seems unrelated to this issue but perhaps my Upnp install is junk?
```
[  5%] Building CXX object CMakeFiles/libgerbera.dir/src/autoscan_inotify.cc.o
In file included from /Users/gerbera/src/action_request.cc:32:
In file included from /Users/gerbera/src/action_request.h:36:
In file included from /usr/local/include/upnp-1.8/upnp.h:47:
In file included from /usr/local/include/upnp-1.8/UpnpInet.h:41:
/usr/include/arpa/inet.h:77:13: error: declaration of 'inet_ntop' has a different language linkage
const char      *inet_ntop(int, const void *, char *, socklen_t);
                 ^
/System/Library/Frameworks/Kernel.framework/Headers/netinet/in.h:649:20: note: previous declaration is here
extern const char *inet_ntop(int, const void *, char *, socklen_t); /* in libkern*/
                   ^
```

--- a/README.md
+++ b/README.md
@@ -108,6 +108,15 @@ make -j4
 sudo make install
 ````
 
+#### On macOS
+
+_The following has been tested on macOS Sierra 10.12.6_
+
+1. Install prerequisites using homebrew
+1. Install `ossp-uuid` using homebrew
+1. Install `pupnp18` and `duktape` using the scripts provided.
+1. Use the cmake option `-DCMAKE_FIND_FRAMEWORK=LAST` to avoid Clang linkage errors
+
 
 ### Quick start build instructions:
 

--- a/scripts/install-duktape.sh
+++ b/scripts/install-duktape.sh
@@ -10,4 +10,11 @@ fi
 wget http://duktape.org/duktape-2.1.0.tar.xz
 tar -xJvf duktape-2.1.0.tar.xz
 cd duktape-2.1.0
+
+if [ "$unamestr" == 'Darwin' ]; then
+  # Patch Makefile to install on macOS
+  # macOS does not support -soname, replace with -install_name
+  sed -i -e 's/-soname/-install_name/g' Makefile.sharedlibrary
+fi
+
 $makeCMD -f Makefile.sharedlibrary && sudo $makeCMD -f Makefile.sharedlibrary install

--- a/scripts/install-pupnp18.sh
+++ b/scripts/install-pupnp18.sh
@@ -1,11 +1,13 @@
 #!/bin/sh
 set -ex
 
-extraFlags="--prefix=/usr"
-
 unamestr=`uname`
 if [ "$unamestr" == 'FreeBSD' ]; then
    extraFlags=""
+elif [ "$unamestr" == 'Darwin' ]; then
+   extraFlags="--prefix=/usr/local"
+else
+   extraFlags="--prefix=/usr"
 fi
 
 wget https://github.com/mrjimenez/pupnp/archive/release-1.8.1.tar.gz -O pupnp-1.8.1.tgz

